### ISSUE TITLE
Do you plan to support function replacer in JSON.stringify?

### ISSUE THREAD

**Issue by ayahito-saji** [NeilFraser/JS-Interpreter#266]
It seems that js-interpreter currently does not support functions for JSON.stringify replacer.
Are there any plans for support in the future?

ECMAScript 5.1 does support them, as seen at https://262.ecma-international.org/5.1/#sec-15.12.3
For security reasons, it is difficult to make replacer functions native, so a custom implementation of JSON.stringify will be needed to support function replacer in JSON.stringify.
Considering speed, it would be best to use the custom implementation when replacer is present and use the native implementation when it is not present.

Thank you.

**Comment by NeilFraser** [comment]
I agree with the approach you outline.  It's the same as that used by String's .replace function which has native code wrapped by a polyfill:
https://github.com/NeilFraser/JS-Interpreter/blob/master/interpreter.js#L1759

There are no current plans to support functions in JSON.stringify.  You are the first to ask about it.  Out of curiosity, what are you using JS-Interpreter for?  You are certainly digging deeply in a lot of areas.  :)

**Comment by ayahito-saji** [comment]
Thank you for sharing the code!

I am trying to use [core-js](https://github.com/zloirock/core-js), webpack, and babel to run a TypeScript program that supports ES2015 and later syntax.
Another purpose of this attempt is to test the js-interpreter with core.js, which utilizes many types of syntax.
While much of the code works with this wonderful interpreter, I found that core-js behaved unexpectedly depending on some unsupported syntaxes and unstable behavior.

**Comment by semireg** [comment]
@ayahito-saji, can you offer [any guidance](https://github.com/NeilFraser/JS-Interpreter/issues/272) on how you used webpack to bundle for JS Interpreter?
